### PR TITLE
Hotfix/webdip rr bugs

### PR DESCRIPTION
**Pull request by sariacodazen**
This PR address forced retreat, units being placed in proper territories during retreats phase, and saving orders during forced disband

Test Plan: 
Pull code
Put game in retreats phase with only one option (game should force draw a retreat arrow to that territory)
Put a unit in a position with nowhere to retreat (unit should show disbanded icon and order state in context should show Saved,Completed" and Save button should be unclickable)

### CODE REVIEW

File: beta-src/src/utils/map/getUnits.ts [hunk 1]
```diff
@@ -25,11 +25,16 @@ export default function getUnits(
...
+        )
+      : undefined;
 
     if (
       territoryStatus?.occupiedFromTerrID &&
       unit.id === territoryStatus.unitID &&
-      territoryStatus?.ownerCountryID === unit.countryID &&
+      occupiedTerritory?.ownerCountryID === unit.countryID &&
       territoryHasMultipleUnits.length > 1 &&
       contextVars?.context.phase === "Retreats"
```
Comment (bryancodazen) [inline beta-src/src/utils/map/getUnits.ts:39]: Don't use contextVars for phase data, use the phase in game overview instead.

File: beta-src/src/utils/map/updateUnitsRetreat.ts [hunk 2]
```diff
@@ -20,8 +21,15 @@ export default function updateUnitsRetreat(state): void {
     ordersMeta: OrdersMeta;
     overview: GameOverviewResponse;
   } = current(state);
+  let orderStates;
+
+  if (contextVars) {
+    orderStates = getOrderStates(contextVars.context.orderStatus);
+  }
+
```
Comment (bryancodazen) [inline beta-src/src/utils/map/updateUnitsRetreat.ts:29]: Don't use orderStates, use gameOverview -> user -> member -> orderStatus

File: beta-src/src/components/ui/WDMoveControls.tsx [hunk 3]
```diff
@@ -95,13 +99,20 @@ const WDMoveControls: React.FC<WDMoveControlsProps> = function ({
     }
   };
 
+  const commandActions = {
+    SAVE_ORDERS: (command) => {
+      clickButton(Move.SAVE);
+    },
+  };
```
Comment (bryancodazen) [inline beta-src/src/components/ui/WDMoveControls.tsx:106]: I think you need to delete the command after it is processed. Look at the command actions in WDTerritory, WDMapController, or WDUnitController as examples.

**Review by bryancodazen — CHANGES_REQUESTED** [review]

File: beta-src/src/utils/map/updateUnitsRetreat.ts [hunk 4]
```diff
@@ -47,5 +55,17 @@ export default function updateUnitsRetreat(state): void {
       };
     }
     setCommand(state, command, "unitCommands", unitID);
+
+    if (
+      currentOrders.length === 1 &&
+      type === "Disband" &&
+      !allowedBorderCrossings?.length &&
+      !orderStates?.Completed
```
Comment (mariomelchor) [inline beta-src/src/utils/map/updateUnitsRetreat.ts]: If you get this from state orderStates wouldn't be null.

**Review by mariomelchor — CHANGES_REQUESTED** [review]

**Review by bryancodazen — CHANGES_REQUESTED** [review]
We should not be able to select other units (that don't have retreat orders) during the retreats phase. 

https://user-images.githubusercontent.com/93741108/168933187-3ab98ef0-c2f0-47a1-8465-e1697911317d.mov

**Review by bryancodazen — CHANGES_REQUESTED** [review]
Ran into another issue as well where the unit only has one place to retreat to (Picardy), yet the arrow is not getting drawn (the order is not being set in ordersMeta either).


https://user-images.githubusercontent.com/93741108/168935583-41258669-a920-46ba-bdd5-90e139a958b3.mov

**Review by bryancodazen — APPROVED** [review]
lgtm

**Review by justinyoocodazen — APPROVED** [review]

**Pull request merged**